lib/ecto/adapters/sql.ex

```diff
@@ -1351,22 +1398,24 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
-  defp log_stacktrace(stacktrace, repo) do
-    with [_ | _] <- stacktrace,
-         {module, function, arity, info} <- last_non_ecto(Enum.reverse(stacktrace), repo, nil) do
+  defp log_stacktrace([_ | _] = stacktrace, repo, {module, function, args}) do
+    entries = apply(module, function, args ++ [Enum.reverse(stacktrace), repo])
+
+    for {{module, function, arity, info}, idx} <- Enum.with_index(entries) do
       [
         ?\n,
         IO.ANSI.light_black(),
+        List.duplicate(?\s, 2 * idx),
         "↳ ",
```

Comment:
```suggestion
        if(idx == 0, do: "↳ ", else: "  "),
```

This way we don't print a staircase, WDYT?